solvation_analysis/plotting.py

```diff
@@ -0,0 +1,311 @@
+        elif x_axis == "solution":
+            # each species is a line
+            for property_dict in cleaned_properties:
+                fig.add_trace(go.Scatter(x=list(property_dict.values()), y=list(property_dict.keys())))
+    else:
+        if x_axis == "species":
+            # each solution is a bar
+            for property_dict in cleaned_properties:
+                fig.add_trace(go.Bar(x=list(property_dict.keys()), y=list(property_dict.values())))
+        elif x_axis == "solution":
+            # each species is a bar
+            df = pd.DataFrame(data=properties_dict)
+            cond = [[elem in col for col in df.columns] for elem in df.index]
+            cond = pd.DataFrame(data=cond)
+            df.mask(cond)
+            for i in range(len(df.index)):
+                # TODO iterate through the cols/rows
+                fig.add_trace(go.Bar(x=df[i], y=df.index))
+    return fig
+
+def clean(properties, keep_solvents):
+    cleaned_properties = []
+    for i in range(len(properties)):
+        cleaned_properties.append({})
+        for solvent in properties[i]:
+            if solvent in keep_solvents:
+                cleaned_properties[i][solvent] = properties[i][solvent]
+    return cleaned_properties
+
+
+def catch_different_solvents(solvents):
+    if not all(elem == solvents[0] for elem in solvents):
+        raise ValueError("Solutions must have identical solvent compositions. Make sure that keep_solvents for each"
+            " solution lists only the solvents common to all solutions")
+
+def compare_free_solvents(solutions):
+    # this should be a grouped vertical bar chart or a line chart
+    # 1.0 should be marked and annotated with a dotted line
+    fig = compare_solvent_dicts()
+    return
+
+
+def compare_pairing(solutions, x_axis, series=False, keep_solvents=None):
+    # this should be a grouped vertical bar chart or a line chart
+    # *** should there be another (boolean) parameter that
+    # 1.0 should be marked and annotated with a dotted line
+    """
+    Compares the pairing of multiple solutions.
+    Parameters
+    ----------
+    solutions : a dict of Solution objects (should not be mutable)
+    series : Boolean (False when a line chart is not wanted)
+    keep_solvents : a list of solvent names (strings) to keep in the analysis
+
+    Returns
+    -------
+    fig : Plotly.Figure
+
+    """
+    # you can also do kwargs instead of writing out all the keywords
+    keep_solvents = keep_solvents or []
+    if keep_solvents:
+        solvents_list = []
+        for solution in solutions:
+            temp_solvents = list(solutions[solution].solvents)
+            solvents_list.append(list(set(temp_solvents).intersection(set(keep_solvents))))
+    else:
+        solvents_list = [list(solutions[solution].solvents.keys()) for solution in solutions]
+
+    catch_different_solvents(solvents_list)
```

Comment:
I'd recommend wrapping all of this logic into a single, well-named function. It should take the solvents from each electrolyte (`solvents_list` here) and keep_solvents as arguments and either throw an error or return keep_solvents.

You'll need to repeat this logic for every `compare_*` function, so best to package it up.